agent/engine/docker_container_engine.go

```diff
@@ -136,7 +142,8 @@ func (dg *DockerGoClient) pullImage(image string) DockerContainerMetadata {
 	// Workaround for devicemapper bug. See:
 	// https://github.com/docker/docker/issues/9718
 	pullLock.Lock()
-	defer pullLock.Unlock()
+	ttime.Sleep(utils.AddJitter(0, dockerPullJitter))
```

Comment:
I'm really concerned here that this "workaround" now isn't: we now have a race condition that makes it possible for two pulls with a common layer to occur just starting at different times.